rkyv/src/util/aligned_vec.rs

```diff
@@ -348,6 +348,39 @@ impl AlignedVec {
             }
         }
     }
+    /// Resizes the Vec in-place so that len is equal to new_len.
+    ///
+    /// If new_len is greater than len, the Vec is extended by the difference, with each additional slot filled with value. If new_len is less than len, the Vec is simply truncated.
+    /// 
+    /// # Panics
+    ///
+    /// Panics if the new length exceeds `usize::MAX` bytes.
+    ///
+    /// # Examples
+    /// ```
+    /// use rkyv::AlignedVec;
+    ///
+    /// let mut vec = AlignedVec::new();
+    /// vec.resize(3, 2);
+    /// assert_eq!(vec.as_slice(), &[2, 2, 2]);
+    ///
+    /// let mut vec = AlignedVec::new();
+    /// vec.extend_from_slice(&[1, 2, 3, 4]);
+    /// vec.resize(2, 0);
+    /// assert_eq!(vec.as_slice(), &[1, 2]);
+    /// ```
+    #[inline]
+    pub fn resize(&mut self, new_len: usize, value: u8) {
+        let len = self.len;
+        if new_len > len {
+            self.reserve(new_len - len);
+            let slice = unsafe { slice::from_raw_parts_mut(self.ptr.as_ptr(), new_len) };
```

Comment:
I think we should be using `ptr::write_bytes` here since making a slice of the bytes creates a reference to uninitialized memory. `write_bytes` may also result in some more efficient code.

Reply:
fixed